tests/api/test_views.py

```diff
@@ -135,3 +215,1443 @@ def test_retrieve_licence_list_with_correct_api_key(app: DjangoTestApp):
         'id': str(licence.identifier),
         'title': licence.title
     }]
+
+
+@pytest.mark.django_db
+def test_get_all_datasets_without_api_key(app: DjangoTestApp):
+    res = app.get(reverse("api-dataset"), expect_errors=True)
+    assert res.status_code == 403
+
+
+@pytest.mark.django_db
+def test_get_all_datasets(app: DjangoTestApp):
+    domain = Site.objects.get_current().domain
+    dataset = DatasetFactory()
```

Comment:
Add more datasets with different orgs, this api endpoint should only return datasets of representative org and should also return non-public datasets.